### PR TITLE
Fix browser extension broken links when branch is specified

### PR DESCRIPTION
**Pull request by tjkandala**
Fix [#11057](https://github.com/sourcegraph/sourcegraph/issues/11057).

The Sourcegraph browser extension link used to break on the root directory of a repo when a branch was specified. This happened because we would throw an error when the path structure wasn't exactly:
\<empty>/\<user>/\<repo>/(blob | tree)/\<commitID>/\<path/to/file>

The href for the a.js-permalink-shortcut element on the root directory looks like this: "/sourcegraph/sourcegraph/tree/cff20a845c2263deacdcbf18e3b50e2b3c4af94c" (no path).

I added a check for page type in `getFilePath()` in order to allow tree pages to have an empty path. This resulted in a complication in `resolveFileInfo()` because it prepended all paths with a slash, so I made an exception for falsy strings.

Result:
extension_master

The link shows up now.
extension_branch

The link to the repository works.
result_after_click

### CODE REVIEW

File: browser/src/browser-extension/scripts/contentPage.main.ts [hunk 1]
```diff
@@ -105,3 +105,4 @@ async function main(): Promise<void> {
 }
 
 main().catch(console.error.bind(console))
+
```
Comment (lguychard) [inline browser/src/browser-extension/scripts/contentPage.main.ts:108]:
```suggestion
```

File: browser/src/shared/code-hosts/github/fileInfo.ts [hunk 2]
```diff
@@ -41,7 +41,8 @@ export const resolveFileInfo = (): BlobInfo => {
     const { revisionAndFilePath, rawRepoName } = parsedURL
 
     const filePath = getFilePath()
-    const filePathWithLeadingSlash = filePath.startsWith('/') ? filePath : `/${filePath}`
+    // Don't prepend empty filePath with slash; it's the root directory of the repo
+    const filePathWithLeadingSlash = !filePath ? '' : filePath.startsWith('/') ? filePath : `/${filePath}`
```
Comment (lguychard) [inline browser/src/shared/code-hosts/github/fileInfo.ts:45]:
```suggestion
    const filePathWithLeadingSlash = filePath && !filePath.startsWith('/') ? `/${filePath}` : filePath
```

Nit: I would find this slightly easier to grok with a single ternary operator.

File: dev/postgres_exporter.sh [hunk 3]
```diff
@@ -28,4 +28,4 @@ if [[ "$OSTYPE" == "linux-gnu" ]]; then
   DATA_SOURCE_NAME="postgresql://${PGUSER}:${PGPASSWORD}@${ADJUSTED_HOST}:${PGPORT}/postgres?sslmode=${PGSSLMODE:-disable}"
 fi
 
-exec docker run --rm -p9187:9187 ${NET_ARG} --name="$CONTAINER" -e DATA_SOURCE_NAME="${DATA_SOURCE_NAME}" ${IMAGE}
+exec docker run --rm -p9187:9187 ${NET_ARG} --name="$CONTAINER" -e DATA_SOURCE_NAME="${DATA_SOURCE_NAME}" ${IMAGE}
```
Comment (lguychard) [inline dev/postgres_exporter.sh:31]: You probably didn't mean to change this? 🙂

File: shared/src/api/client/connection.ts [hunk 4]
```diff
@@ -69,7 +69,6 @@ export async function createExtensionHostClientConnection(
 
     /** Proxy to the exposed extension host API */
     const initializeExtensionHost = comlink.wrap<ExtensionHostAPIFactory>(endpoints.proxy)
-
```
Comment (felixfbecker) [inline shared/src/api/client/connection.ts:72]: Unintentional change?

Comment (tjkandala) [reply shared/src/api/client/connection.ts:72]: Yeah, I'm not sure what happened 🤔

**Comment by tjkandala** [comment]
Thanks for the feedback! I just noticed that the button still links to the master branch even when a branch is specified, so I'll look into that.

**Comment by tjkandala** [comment]
The ViewOnSourcegraphButton links to the right branch now. When there is no path, don't slice `revisionAndFilePath`.

fix1

fix2

**Pull request merged**